queue_job/job.py

```diff
@@ -300,7 +300,7 @@ def job_record_with_same_identity_key(self):
             .search(
                 [
                     ("identity_key", "=", self.identity_key),
-                    ("state", "in", [PENDING, ENQUEUED]),
+                    ("state", "in", [WAIT_DEPENDENCIES, PENDING, ENQUEUED, STARTED]),
```

Comment:
Wondering: shall we make optional to look for failed jobs too?
Example: I have some jobs to send out files. A cron is responsible to spawn jobs whenever a record needs to send out something. When jobs fail (eg: external service is off) they will get re-created because `FAILED` is not considered here.

Reply:
Actually the right approach seems to be to raise a RetryableJobError that will set the state as pending